### PR TITLE
feat: Add support for Compass on Wasm

### LINKED ISSUE #11496: Add support for `Compass`
### Current behavior

Not implemented

### Expected behavior

Implemented

### How to reproduce it (as minimally and precisely as possible)

_No response_

### Workaround

_No response_

### Works on UWP/WinUI

Yes

### Environment

Uno.WinUI / Uno.WinUI.WebAssembly / Uno.WinUI.Skia

### NuGet package version(s)

_No response_

### Affected platforms

Android, iOS, Mac Catalyst, Skia (WPF), Skia (GTK on Linux/macOS/Windows), Skia (Linux Framebuffer), Skia (Tizen)

### IDE

Visual Studio 2022

### IDE version

_No response_

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by morning4coffe-dev**
GitHub Issue (If applicable): closes #11496 

## PR Type

What kind of change does this PR introduce?

- Feature

-->

## What is the current behavior?
`Compass` is currently not supported on Wasm.

## What is the new behavior?
Added support for `Compass` sensor.

### <samp>🤖 Generated by Copilot at 3e223f2</samp>

This pull request adds support for the `Compass` class on WebAssembly (WASM) by implementing a native version using the `Accelerometer` and `Magnetometer` sensors. It also updates the documentation and the conditional compilation directives to reflect the new feature.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

File: src/Uno.UWP/Devices/Sensors/Compass.wasm.cs [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+		return null;
+	}
+
+	private void StartReadingChanged()
+	{
+		_accelerometer = Accelerometer.GetDefault();
+		_magnetometer = Magnetometer.GetDefault();
+
+		if (_accelerometer != null && _magnetometer != null)
```
Comment (kazo0) [inline src/Uno.UWP/Devices/Sensors/Compass.wasm.cs:68]: We won't subscribe to ReadingChanged for the accelerometer if the magnetometer is null? Or vice-versa?

Comment (kazo0) [reply src/Uno.UWP/Devices/Sensors/Compass.wasm.cs:68]: Oh nevermind, I see :)

File: src/Uno.UWP/Devices/Sensors/Compass.wasm.cs
Comment (kazo0) [inline src/Uno.UWP/Devices/Sensors/Compass.wasm.cs:1]: Do we have runtime tests that this will already cover?

**Review by MartinZikmund — APPROVED** [review]

File: src/Uno.UWP/Devices/Sensors/Compass.wasm.cs
Comment (MartinZikmund) [reply src/Uno.UWP/Devices/Sensors/Compass.wasm.cs:1]: Unfortunately the sensors are not really testable in CI...

**Pull request merged**